### PR TITLE
BAPL-1167/JBPM-8123 - DMN Support

### PR DESCRIPTION
**Pull request by Josephblt**
https://issues.jboss.org/browse/BAPL-1167
https://issues.jboss.org/browse/JBPM-8123

New properties for Businnes Rule Task:
Rule Language (DRM or DMN)
Namespace
Decision Name
DMN Model Name

Conditional UI. Depending on the Rule Language of choice, only
related properties are shown.

### CODE REVIEW

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Hey @romartin, @jomarko, @etirelli. If you guys could please test this, I'd be glad. Thanks.

**Comment by Josephblt** [comment]
Full downstream build failed due to unrelated changes. So we can go ahead with the testing. Thanks.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/filters/RuleLanguageFilterProvider.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        return Arrays.asList(ruleFlowGroupFilter, namespaceFilter, decisionNameFilter, dmnModelNameFilter);
+    }
+
+    void onFormFieldChanged(@Observes FormFieldChanged formFieldChanged) {
+        final String ruleLanguageFieldName = BusinessRuleTask.EXECUTION_SET + "." + BusinessRuleTaskExecutionSet.RULE_LANGUAGE;
+        if (!Objects.equals(formFieldChanged.getName(), ruleLanguageFieldName)) {
+            return;
+        }
+
+        refreshFormPropertiesEvent.fire(new RefreshFormPropertiesEvent(sessionManager.getCurrentSession(), formFieldChanged.getUuid()));
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/filters/RuleLanguageFilterProvider.java]: good @tiagodolphine  @Josephblt  :+1:

**Comment by jomarko** [comment]
The validation says I didn't fill some values which are actually filled.

screenshot from 2019-01-18 09-35-53

**Comment by jomarko** [comment]
@Josephblt also one more question what do you think, currently there is selector with options **Select**, **DRL**, and **DMN**, however first two options seems to me are the same, just asking for your opinion, shouldn't we merge first two options?

**Comment by Josephblt** [comment]
Hey @jomarko. Yes I think we can merge this options. Also I did not notice the validation issues. Let you know when they are fixed.

**Comment by romartin** [comment]
@Josephblt how's progress on this?

**Comment by Josephblt** [comment]
Hello @romartin. All set. Ready for review. @jomarko, @hasys if you guys could please take a look, that would be great. Thanks.

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Hey @jomarko  @hasys 
Please this task is starting to be urgent, can you please do the review ASAP? Thanks in advance!

**Comment by jomarko** [comment]
@Josephblt thank you, @romartin I did review, from my point of view this can be merged because user is able t model process that invokes DMN model, however I think the linked BAPL jira can not be marked as resolved, see this [issue](https://issues.jboss.org/browse/JBPM-8172)

**Pull request merged**